python/jiminy_py/src/jiminy_py/simulator.py

```diff
@@ -436,16 +457,25 @@ def simulate(self,
                                   if available.
                                   Optional: None by default.
         """
+        q_init, v_init, a_init, is_state_theoretical = single_to_multi(self.engine,
+            q_init, v_init, a_init, is_state_theoretical)
+        
+        for system, name in zip(self.engine.systems, self.engine.systems_names):
```

Comment:
what about making yet another helper method for this. You could even gather this with `single_to_multi` and call it instead `sanitize_state` for instance.